### PR TITLE
Fixes off-by-one error and changes drop_last for Cracks

### PR DESCRIPTION
**Pull request by harukafukukawa**
- The off-by-one error in make_dataset in the VideoLoader was causing us to drop the last sample of our datasets.
	- Increases samples available for the training set from 90 -> 91
	- Increases samples available for the testing set from 10 -> 11
- Changing drop_last=True to False allows for the full dataset to be utilized for training.
	- Increases training set from 80 -> 91
	- Increases test set from 8 -> 11
- Since our cracks dataset is small, I think it would be best to use the full training set for each epoch.

### CODE REVIEW

**Comment by redjumper25** [comment]
These changes look good. The `make_dataset` change is valid. Regarding the `drop_last=True` however, I believe the data is still being used by the models. Since our sequences are shuffled in the batches, `drop_last` batch removal shouldn't be significant over 200 epochs. I think the only "noticeable" item it changes is the number of output sequences on inference.

Some things to observe when changing the `drop_last` to `False`: how are the models implementing any form of batch normalization? By inputting a smaller batch size, are we inputting data that is significantly more adjusted than the other batches? How is this going to impact the gradient updates at the end of the batch?

It's definitely worth testing to see how the results change, in the future it might be worthwhile to consider adding this as an argument to the parser so that it can be changed by command line.

**Review by redjumper25 — APPROVED** [review]

**Comment by harukafukukawa** [comment]
> Regarding the `drop_last=True` however, I believe the data is still being used by the models. Since our sequences are shuffled in the batches, `drop_last` batch removal shouldn't be significant over 200 epochs. I think the only "noticeable" item it changes is the number of output sequences on inference.

Yes good point! You're right the data still gets used by the models, after multiple epochs, but the full dataset isn't used in any single epoch. It's a good point though, so I tested setting `drop_last`=False and True for the training data loader for SimVP and ConvLSTM.
In both models, the MSE decreased when setting drop_last=False by 8 and 7 (ie, a *tiny* improvement). I think this is probably because it ends up doing one additional gradient update every epoch rather than because we include the last partial batch. I think it'll make more intuitive sense to increase the number of epochs rather than rely on the (possibly unpredictable) effect of setting `drop_last=False`, so I'll revert that change for the training dataloader.

> Some things to observe when changing the `drop_last` to `False`: how are the models implementing any form of batch normalization? By inputting a smaller batch size, are we inputting data that is significantly more adjusted than the other batches? How is this going to impact the gradient updates at the end of the batch?

I don't think the smaller batch size would impact our results much because our dataset is pretty similar throughout, and if we follow the convention of using batch sizes that are powers of two for our 91 samples, we never end up with the last batch being significantly smaller than the other batches--- but this is feeling hand wavy, so aiming for consistency and dropping the last partial batch seems like to would be better 👍 
 
> It's definitely worth testing to see how the results change, in the future it might be worthwhile to consider adding this as an argument to the parser so that it can be changed by command line.

👍 

tldr; I'll change the drop_last back to True for the training dataloader. Thanks for the thoughtful feedback! 😄
